### PR TITLE
feat: TS - Split Sender and Receiver 

### PR DESCRIPTION
**Pull request by newfla**
The Channel management on the TS side is now aligned with that already present in Rust with the channel divided into two entities: Sender and Receiver

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,48 +80,37 @@ fn example_function (app_handle: AppHandle) {
...
+import {channel} from "tauri-plugin-channel-api"
+import type {Sender, Receiver} from "tauri-plugin-channel-api"
 
-let channel = await invoke<Channel>('example_function')
-receive(channel, (event) => console.log(event));
-emit(channel, "1");
+type BackendEvents = { Progress: number } 
+
+[sender, receiver]  = await channel('example_function')
+receiver.listen<BackendEvents>(event => console.log(JSON.stringify(event)));
```
Comment (brainrepo) [inline README.md:98]: I recall that you said the an unlisten function is not needed, Am I wrong?

File: package.json [hunk 2]
```diff
@@ -1,25 +1,30 @@
 {
-  "name": "tauri-plugin-channel-api",
+  "name": "@fourviere/tauri-plugin-channel-manifest",
```
Comment (brainrepo) [inline package.json:2]: why  "-manifest"?

File: packages/examples/tauri-app/src/lib/FastBar.svelte [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    await receiver.listen<ProgressBackendEvents>((event) => {
+      msg = JSON.stringify(event);
+      if (msg === '"Stopped"' || msg === '"Done"') {
+        start_button_enabled = true;
+        stop_button_enabled = false;
+      }
+    });
+  }
+
```
Comment (brainrepo) [inline packages/examples/tauri-app/src/lib/FastBar.svelte:28]: is there an unlisten function?

**Review by brainrepo — DISMISSED** [review]
LGTM, I just left some comments

File: package.json (hunk 2 above)
Comment (newfla) [reply package.json]: Oh, the infamous typo =D
Fixed, thx

File: README.md (hunk 1 above)
Comment (newfla) [reply README.md:98]: Ok, let me clarify this point:
When _all rust senders_ are dropped an "Unlisten" message is sent and the _ts receiver_ unmount itself and there is no need to call receiver.unlisten()
Unfortunately there is no way (or at least I don't see one) to enforce that senders are going to be dropped and therefore I've made `receiver::unlisten` public

File: packages/examples/tauri-app/src/lib/FastBar.svelte (hunk 3 above)
Comment (newfla) [reply packages/examples/tauri-app/src/lib/FastBar.svelte:28]: See first comment discussion

**Pull request merged**